Data/Repositories/AccountRepository.cs

```diff
@@ -90,4 +91,30 @@ public async Task WithdrawAccountAsync(int accountId, decimal amount)
         account.Balance = currentBalance - amount;
         await _context.SaveChangesAsync();
     }
+
+    public async Task TransferAsync(int fromAccountId, int toAccountId, decimal amount)
+    {
+        if (amount <= 0)
+            throw new ArgumentException("Amount must be greater than zero.", nameof(amount));
+
+        // Загружаем оба счёта
+        var fromAccount = await _context.Accounts.FindAsync(fromAccountId);
+        var toAccount = await _context.Accounts.FindAsync(toAccountId);
+
+        if (fromAccount == null || toAccount == null)
+            throw new KeyNotFoundException("One or both accounts not found.");
+
+        var fromBalance = fromAccount.Balance ?? 0m;
+        if (fromBalance < amount)
+            throw new InvalidOperationException("Insufficient funds on source account.");
+
+        // Транзакция на уровне БД
+        using var transaction = await _context.Database.BeginTransactionAsync();
```

Comment:
it's very good that you use transaction here, but since it's asynchronous don't forget to add await before using
```suggestion
        await using var transaction = await _context.Database.BeginTransactionAsync();
```